compiler/rustc_hir_typeck/src/check.rs

```diff
@@ -108,7 +121,13 @@ pub(super) fn check_fn<'a, 'tcx>(
         hir::FnRetTy::DefaultReturn(_) => body.value.span,
         hir::FnRetTy::Return(ty) => ty.span,
     };
+
     fcx.require_type_is_sized(declared_ret_ty, return_or_body_span, traits::SizedReturnType);
+    // We checked the root's signature during wfcheck, but not the child.
+    if fcx.tcx.is_typeck_child(fn_def_id.to_def_id()) {
```

Comment:
I'm assuming doing this for all functions is too restrictive wrt cycles (and a breaking change on top?)

Reply:
No it just leads to really annoying diagnostic duplication

Reply:
We already check that the args of a function signature are WF, so it's not a breaking change

Reply:
so... we could remove it from wf checking and just do it here unconditionally?

Reply:
Oh, yeah I guess. What's the status on wfchecking and hir typecking affecting each other? Does wfcheck not keep typeck from running anymore?

Reply:
yea, there's no more stops until the "avoid running lints if analyses errored"

Reply:
Oh, one more weirdness is that we still need to run this on fn items that dont have bodies during wfcheck.

In general, I think this split makes sense (at least unless/until we get rid of wfcheck as a pass completely) since the only reason we delay these checks for non-root items is bc their params/return can have infer vars, while the root items cannot